packages/flutter/lib/src/material/text_form_field.dart

```diff
@@ -309,21 +315,27 @@ class TextFormField extends FormField<String> {
 }
 
 class _TextFormFieldState extends FormFieldState<String> {
-  TextEditingController? _controller;
+  late final RestorableTextEditingController _controller = RestorableTextEditingController(text: widget.initialValue);
```

Comment:
We should only have this instantiated if we actually need it, similarly to how we do it in the Textfield itself: https://github.com/flutter/flutter/pull/63401/files